File: app/src/main/java/cn/ac/ict/cana/newversion/modules/count/CountSimKeyboardActivity.java
package cn.ac.ict.cana.newversion.modules.count;

import android.app.Activity;
import android.content.Context;
import android.content.DialogInterface;
import android.content.Intent;
import android.content.SharedPreferences;
import android.graphics.Point;
import android.media.AudioManager;
import android.media.SoundPool;
import android.os.Bundle;
import android.support.v7.app.AlertDialog;
import android.util.Log;
import android.view.Gravity;
import android.view.MenuItem;
import android.view.View;
import android.view.inputmethod.EditorInfo;
import android.widget.Button;
import android.widget.GridLayout;
import android.widget.TextView;
import android.widget.Toast;

import com.lovearthstudio.duasdk.Dua;

import java.io.BufferedWriter;
import java.io.File;
import java.io.FileWriter;
import java.io.IOException;
import java.util.ArrayList;
import java.util.HashMap;
import java.util.Map;

import cn.ac.ict.cana.R;
import cn.ac.ict.cana.helpers.DataBaseHelper;
import cn.ac.ict.cana.helpers.ModuleHelper;
import cn.ac.ict.cana.newversion.activities.FeedBackActivity;
import cn.ac.ict.cana.newversion.mode.CountData;
import cn.ac.ict.cana.newversion.mode.History;
import cn.ac.ict.cana.newversion.modules.guide.ModelGuideActivity2;
import cn.ac.ict.cana.newversion.provider.HistoryProvider;
import cn.ac.ict.cana.newversion.utils.FileUtils;

/**
 * Created by zhongxi on 2016/10/19.
 */
public class CountSimKeyboardActivity extends Activity {

    private String randomStr;
    private String version;
    private TextView nextet;
    private Button nextbtn;
    private Intent intent;
    private int times;
    private boolean isRight;
    private ArrayList<String> result;

    private GridLayout gridLayout;
    private TextView tv;
//    private Button confirmBtn;

    private String[] chars;

    private int[] source;
    private SoundPool pool;
    private Map<String, Integer> poolMap;
    private boolean isLoad;
    public boolean isMusic;
    private Button musicBtn;
    private boolean isNotFull;
    private int grade;
    private TextView tvGrade;
    private int inputNum = 2;
    private int maxTestNum = 6;
    private CountData countData;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_count_simkeyboard);

        init();
    }

    public void init() {

        isMusic = true;
        isNotFull = true;
        pool = new SoundPool(11, AudioManager.STREAM_MUSIC, 0);
        poolMap = new HashMap<>();
        source = new int[]{
                R.raw.counts0,
                R.raw.counts1,
                R.raw.counts2,
                R.raw.counts3,
                R.raw.counts4,
                R.raw.counts5,
                R.raw.counts6,
                R.raw.counts7,
                R.raw.counts8,
                R.raw.counts9,
                R.raw.counts_del
        };

        for (int i = 0; i < 11; i++) {
            //   poolMap.put("index" + i, pool.load(this, source[i], 1));
        }
        pool.setOnLoadCompleteListener(new SoundPool.OnLoadCompleteListener() {
            @Override
            public void onLoadComplete(SoundPool soundPool, int sampleId, int status) {

                if (sampleId == poolMap.size()) {
                    isLoad = true;
                }
            }
        });

        musicBtn = (Button) findViewById(R.id.count_voice);
        musicBtn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                if (isMusic) {
                    musicBtn.setBackgroundResource(R.drawable.count_keymusic_close);
                    isMusic = false;
                } else {
                    musicBtn.setBackgroundResource(R.drawable.count_keymusic_open);
                    isMusic = true;
                }
            }
        });


        chars = new String[]{
                "7", "8", "9",
                "6", "5", "4",
                "3", "2", "1",
                "0", getApplicationContext().getString(R.string.count_sim_clear), getApplication().getString(R.string.count_sim_delete)
        };

        Point size = new Point();
        getWindowManager().getDefaultDisplay().getSize(size);
        int screenWidth = size.x;
        int oneQuarterWidth = (int) (screenWidth * 0.30);

        intent = this.getIntent();
        randomStr = intent.getStringExtra("data");
        version = intent.getStringExtra("version");
        grade = intent.getIntExtra("grade", 3);
        randomStr = randomStr.substring(0, grade);
        tvGrade = (TextView) findViewById(R.id.tv_grade);
        tvGrade.setText("请输入刚才屏幕上出现的" + grade + "个数字");
        nextet = (TextView) findViewById(R.id.count_simkeyboard_tv);
        nextet.setInputType(EditorInfo.TYPE_CLASS_PHONE);
        nextbtn = (Button) findViewById(R.id.count_simkeyboard_confirmBtn);

        countData = new CountData();
        // 确定按钮点击
        nextbtn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                String str = nextet.getText().toString().trim();
                result.add(str);
                if (str.equals(randomStr)) {
                    isRight = true;
                    if (times == 0) {
                        countData.answer = str;
                    } else {
                        countData.reply = str;
                    }
                    if (grade < 6) {
                        new AlertDialog.Builder(CountSimKeyboardActivity.this).setTitle("提示！").setMessage("恭喜，回答正确!难度升级，下 将进 " + (grade + 1) + "个数字的 记忆游戏!").setNegativeButton("取消", null).setPositiveButton("确定", new DialogInterface.OnClickListener() {
                            @Override
                            public void onClick(DialogInterface dialog, int which) {
                                saveAndContinue();
                            }
                        }).show();
                    } else {
                        saveAndContinue();
                    }
                } else {
                    if (!isRight) {
                        times++;
                        new AlertDialog
                                .Builder(CountSimKeyboardActivity.this)
                                .setTitle("提示！")
                                .setMessage("回答错误！，您还有一次机会")
                                .setPositiveButton("确定", null)
                                .show();
                        if (times >= inputNum) {
                            new AlertDialog
                                    .Builder(CountSimKeyboardActivity.this)
                                    .setTitle("提示！")
                                    .setMessage("很遗憾，回答错误!再来 次吧，下 将进  " + grade + "个数字的 记忆游戏!")
                                    .setNegativeButton("取消", null)
                                    .setPositiveButton("确定", new DialogInterface.OnClickListener() {
                                        @Override
                                        public void onClick(DialogInterface dialog, int which) {
                                            nextTest();
                                        }
                                    })
                                    .show();
                        }
                    }
                }


            }
        });

        result = new ArrayList<>();

//        confirmBtn = (Button)findViewById(R.id.count_simkeyboard_confirmBtn);
        tv = (TextView) findViewById(R.id.count_simkeyboard_tv);
        gridLayout = (GridLayout) findViewById(R.id.coutn_gridlayout_root);
        for (int i = 0; i < chars.length; i++) {
            Button btn = new Button(this);
//            btn.setTextColor(getResources().getColor(R.color.freebie_9));
            btn.setBackgroundResource(R.drawable.count_key_button_style);

            btn.setText(chars[i]);
            if (i > 9) {
                btn.setTextSize(30);
            } else {
                btn.setTextSize(40);
            }
//            btn.setPadding(5,35,5,35);
            btn.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View arg0) {
                    Button btn = (Button) arg0;

                    String btnText = btn.getText().toString().trim();

                    if (!isNotFull && btnText.length() < 2) {

                        //Toast.makeText(CountSimKeyboardActivity.this, getString(R.string.input_full_hint), Toast.LENGTH_SHORT).show();
                        Toast.makeText(CountSimKeyboardActivity.this, "有效数字是" + grade + "位数，您的输入已满。\n请点击确定验证答案或删除重新输入。", Toast.LENGTH_SHORT).show();
//                        SweetAlertDialog sweetAlertDialog = new SweetAlertDialog(CountSimKeyboardActivity.this, SweetAlertDialog.WARNING_TYPE)
//                                .setTitleText(getString(R.string.input_hint))
//                                .setContentText(getString(R.string.input_full_hint))
//                                .setConfirmText(getString(R.string.btn_confirm))
//                                .setConfirmClickListener(new SweetAlertDialog.OnSweetClickListener() {
//                                    @Override
//                                    public void onClick(SweetAlertDialog sDialog) {
//                                        sDialog.dismissWithAnimation();
////                                    sDialog.cancel();
//                                    }
//                                });
//                        sweetAlertDialog.show();
                        return;
                    }

                    if (isMusic && isNotFull) {
                        if (btnText.length() < 2) {
                            //       pool.play(poolMap.get("index" + String.valueOf(btnText)), 1.0f, 1.0f, 0, 1, 1.0f);
                        } else {
                            //      pool.play(poolMap.get("index" + 10), 1.0f, 1.0f, 0, 1, 1.0f);
                        }
                    }
                    if (isMusic && !isNotFull) {
                        //   pool.play(poolMap.get("index" + 10), 1.0f, 1.0f, 0, 1, 1.0f);
                    }

                    String str = tv.getText().toString();
                    if (btnText.equals("删除") || btnText.equals("delete")) {
                        deleteText(arg0);
                        isNotFull = true;
                    } else if (btnText.equals("清空") || btnText.equals("clear")) {
                        clearText(arg0);
                        isNotFull = true;
                    } else {

                        btnText = str.concat(btnText);
                        tv.setText(btnText);
                        if (btnText.length() > grade - 1) {
                            isNotFull = false;
                        } else {
                            isNotFull = true;
                        }
                    }
                }

            });
            GridLayout.Spec rowSpec = GridLayout.spec(i / 3);
            GridLayout.Spec colSpec = GridLayout.spec(i % 3);
            GridLayout.LayoutParams params = new GridLayout.LayoutParams(rowSpec, colSpec);
            params.setGravity(Gravity.FILL);
            gridLayout.addView(btn, params);
            params.width = oneQuarterWidth;
        }

    }

    private void nextTest() {
        FileUtils.countDataList.add(countData);
        Intent intent = new Intent(CountSimKeyboardActivity.this, CountGameActivity.class);
        intent.putExtra("grade", grade);
        startActivity(intent);
        finish();
    }

    protected void saveAndContinue() {
        String content = randomStr;
        if (isRight) {
            content += ";1";
        } else {
            content += ";0";
        }
        if (version.equals("sound")) {
            content += ";1";
        } else {
            content += ";0";
        }
        for (String x : result) {
            content += ";" + x;
        }

        // saveToStorage(content);
        // saveData(randomStr);
        grade++;
        if (grade <= maxTestNum) {
            nextTest();
        } else {
            FileUtils.countDataList.add(countData);
            Intent intent = new Intent(CountSimKeyboardActivity.this, FeedBackActivity.class);
            intent.putExtra("modelName", ModuleHelper.MODULE_COUNT);
            startActivity(intent);
            finish();
        }
    }

    public void saveData(String data) {
        FileUtils.writeData(data + "\n");
    }

    public void saveToStorage(String content) {
        SharedPreferences sharedPreferences = getSharedPreferences("Cana", Context.MODE_PRIVATE);
//        String uuid = sharedPreferences.getString("selectedUser", "None");
//        HistoryProvider historyProvider = new HistoryProvider(DataBaseHelper.getInstance(this));
//        History history = new History(this, uuid, ModuleHelper.MODULE_COUNT);

        // Example: How to write data to file.
        String filePath = History.getFilePath(this, ModuleHelper.MODULE_COUNT);
        File file = new File(filePath);
        try {
            FileWriter fileWrite = new FileWriter(file, true);
            BufferedWriter bufferedWriter = new BufferedWriter(fileWrite);

            bufferedWriter.write(content);

            //Important! Have a new line in the end of txt file.
            bufferedWriter.newLine();
            bufferedWriter.close();
            fileWrite.close();
        } catch (IOException e) {
            Log.e("ExamAdapter", e.toString());
        }

//        history.id = historyProvider.InsertHistory(history);
        SharedPreferences.Editor editor = sharedPreferences.edit();
        editor.putString("HistoryFilePath", filePath);
        editor.apply();
//        Log.d("CountSaveToStorage", String.valueOf(history.id));
        //  EventBus.getDefault().post(new NewHistoryEvent());

        HistoryProvider historyProvider = new HistoryProvider(DataBaseHelper.getInstance(this));
        History history = new History(Dua.getInstance().getCurrentDuaId(), ModuleHelper.MODULE_COUNT, filePath, String.valueOf(grade));
        historyProvider.InsertHistory(history);
    }

    public void clearText(View v) {
        tv.setText("");
    }

    public void deleteText(View v) {
        String str = tv.getText().toString().trim();
        if (str.equals("")) {
            return;
        }
        str = str.substring(0, str.length() - 1);
        tv.setText(str);
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        switch (item.getItemId()) {
            case android.R.id.home:
                startActivity(new Intent(CountSimKeyboardActivity.this, CountMainActivity.class));
                finish();
                break;
        }
        return super.onOptionsItemSelected(item);
    }

    @Override
    protected void onDestroy() {
        if (pool != null) {
            pool.release();
            pool = null;
        }

        super.onDestroy();
    }

    public void next(View view) {
        startActivity(new Intent(this, ModelGuideActivity2.class));
        finish();
    }

    @Override
    public void onBackPressed() {
        startActivity(new Intent(this, CountMainActivity.class));
        finish();
    }
}

File: app/src/main/java/cn/ac/ict/cana/newversion/modules/count/CountSimKeyboardActivity.kt
package cn.ac.ict.cana.newversion.modules.count

import android.content.ContentValues
import android.content.DialogInterface
import android.content.Intent
import android.graphics.Point
import android.media.AudioManager
import android.media.SoundPool
import android.os.Bundle
import android.support.v7.app.AlertDialog
import android.view.Gravity
import android.view.MenuItem
import android.view.View
import android.view.inputmethod.EditorInfo
import android.widget.Button
import android.widget.GridLayout
import android.widget.TextView
import android.widget.Toast
import cn.ac.ict.cana.R
import cn.ac.ict.cana.newversion.helpers.ModuleHelper
import cn.ac.ict.cana.newversion.base.YouMengBaseActivity
import cn.ac.ict.cana.newversion.contant.GlobleData
import cn.ac.ict.cana.newversion.db.database
import cn.ac.ict.cana.newversion.mode.CountData
import cn.ac.ict.cana.newversion.modules.guide.ModelGuideActivity
import cn.ac.ict.cana.newversion.modules.guide.ModelGuideActivity2
import cn.ac.ict.cana.newversion.modules.upload.UploadActivity
import cn.ac.ict.cana.newversion.pagers.ExamPageFragment
import cn.ac.ict.cana.newversion.utils.FileUtils
import com.alibaba.fastjson.JSON
import com.lovearthstudio.duasdk.Dua
import com.lovearthstudio.duasdk.util.encryption.MD5
import kotlinx.android.synthetic.main.activity_count_simkeyboard.*
import org.jetbrains.anko.doAsync
import org.json.JSONObject
import java.util.*

/**
 * 数字记忆录入模块
 */
open class CountSimKeyboardActivity : YouMengBaseActivity() {

    private var randomStr: String? = null
    private var version: String? = null
    private var nextet: TextView? = null
    private var times: Int = 0
    private var isRight: Boolean = false
    private var result: ArrayList<String>? = null

    private var chars: Array<String>? = null

    private var source: IntArray? = null
    private var pool: SoundPool? = null
    private var poolMap: Map<String, Int>? = null
    private var isLoad: Boolean = false
    var isMusic: Boolean = false
    private var musicBtn: Button? = null
    private var isNotFull: Boolean = false
    private var grade: Int = 0
    private val inputNum = 2
    private val maxTestNum = 6
    private var countData: CountData? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_count_simkeyboard)

        init()
    }

    fun init() {

        isMusic = true
        isNotFull = true
        pool = SoundPool(11, AudioManager.STREAM_MUSIC, 0)
        poolMap = HashMap()
        source = intArrayOf(R.raw.counts0, R.raw.counts1, R.raw.counts2, R.raw.counts3, R.raw.counts4, R.raw.counts5, R.raw.counts6, R.raw.counts7, R.raw.counts8, R.raw.counts9, R.raw.counts_del)

        for (i in 0..10) {
            //   poolMap.put("index" + i, pool.load(this, source[i], 1));
        }
        pool!!.setOnLoadCompleteListener { soundPool, sampleId, status ->
            if (sampleId == poolMap!!.size) {
                isLoad = true
            }
        }

        count_voice!!.setOnClickListener {
            if (isMusic) {
                musicBtn!!.setBackgroundResource(R.drawable.count_keymusic_close)
                isMusic = false
            } else {
                musicBtn!!.setBackgroundResource(R.drawable.count_keymusic_open)
                isMusic = true
            }
        }

        chars = arrayOf("1", "2", "3", "4", "5", "6", "7", "8", "9", "0", applicationContext.getString(R.string.count_sim_clear), application.getString(R.string.count_sim_delete))

        val size = Point()
        windowManager.defaultDisplay.getSize(size)
        val screenWidth = size.x
        val oneQuarterWidth = (screenWidth * 0.30).toInt()

        randomStr = intent.getStringExtra("data")
        version = intent.getStringExtra("version")
        grade = intent.getIntExtra("grade", 3)
        randomStr = randomStr!!.substring(0, grade)
        tv_grade!!.text = "请输入刚才屏幕上出现的" + grade + "个数字"
        count_simkeyboard_tv.inputType = EditorInfo.TYPE_CLASS_PHONE

        countData = CountData()
        // 确定按钮点击
        count_simkeyboard_confirmBtn!!.setOnClickListener {
            val str = count_simkeyboard_tv.text.toString().trim { it <= ' ' }
            result!!.add(str)
            if (str == randomStr) {
                isRight = true
                if (times == 0) {
                    countData!!.answer = str
                } else {
                    countData!!.reply = str
                }
                if (grade < 6) {
                    AlertDialog.Builder(this@CountSimKeyboardActivity).setTitle("提示！").setMessage("恭喜，回答正确!难度升级，下 将进 " + (grade + 1) + "个数字的 记忆游戏!").setNegativeButton("取消", null).setPositiveButton("确定") { dialog, which -> saveAndContinue() }.show()
                } else {
                    saveAndContinue()
                }
            } else {
                if (!isRight) {
                    times++
                    AlertDialog.Builder(this@CountSimKeyboardActivity)
                            .setTitle("提示！")
                            .setMessage("回答错误！，您还有一次机会")
                            .setPositiveButton("确定", null)
                            .show()
                    if (times >= inputNum) {
                        AlertDialog.Builder(this@CountSimKeyboardActivity)
                                .setTitle("提示！")
                                .setMessage("很遗憾，回答错误!再来 次吧，下 将进  " + grade + "个数字的 记忆游戏!")
                                .setNegativeButton("取消", null)
                                .setPositiveButton("确定") { dialog, which -> nextTest() }
                                .show()
                    }
                }
            }
        }

        result = ArrayList()

        //        confirmBtn = (Button)findViewById(R.id.count_simkeyboard_confirmBtn);
        for (i in chars!!.indices) {
            val btn = TextView(this)
            btn.setBackgroundResource(R.drawable.count_key_button_bg)
            btn.gravity = Gravity.CENTER
            btn.text = chars!![i]
            btn.setTextColor(resources.getColor(R.color.mainColor))
            if (i > 9) {
                btn.textSize = 30f
            } else {
                btn.textSize = 40f
            }
            btn.setOnClickListener(View.OnClickListener { arg0 ->
                var btnText = btn.text.toString().trim { it <= ' ' }

                if (!isNotFull && btnText.length < 2) {
                    Toast.makeText(this@CountSimKeyboardActivity, "有效数字是" + grade + "位数，您的输入已满。\n请点击确定验证答案或删除重新输入。", Toast.LENGTH_SHORT).show()
                    return@OnClickListener
                }

                val str = count_simkeyboard_tv!!.text.toString()
                if (btnText == "删除" || btnText == "delete") {
                    deleteText(arg0)
                    isNotFull = true
                } else if (btnText == "清空" || btnText == "clear") {
                    clearText(arg0)
                    isNotFull = true
                } else {
                    btnText = str + btnText
                    count_simkeyboard_tv!!.text = btnText
                    isNotFull = btnText.length <= grade - 1
                }
            })
            val rowSpec = GridLayout.spec(i / 3)
            val colSpec = GridLayout.spec(i % 3)
            val params = GridLayout.LayoutParams(rowSpec, colSpec)
            params.setGravity(Gravity.FILL)
            coutn_gridlayout_root.addView(btn, params)
            params.width = oneQuarterWidth
        }

        tv_count_0.setOnClickListener(countKeyClickListener)
        tv_count_1.setOnClickListener(countKeyClickListener)
        tv_count_2.setOnClickListener(countKeyClickListener)
        tv_count_3.setOnClickListener(countKeyClickListener)
        tv_count_4.setOnClickListener(countKeyClickListener)
        tv_count_5.setOnClickListener(countKeyClickListener)
        tv_count_6.setOnClickListener(countKeyClickListener)
        tv_count_7.setOnClickListener(countKeyClickListener)
        tv_count_8.setOnClickListener(countKeyClickListener)
        tv_count_9.setOnClickListener(countKeyClickListener)
    }

    private val countKeyClickListener = View.OnClickListener { p0 ->
        var btn = p0 as TextView
        var btnText = btn.text.toString().trim { it <= ' ' }

        if (!isNotFull && btnText.length < 2) {
            Toast.makeText(this@CountSimKeyboardActivity, "有效数字是" + grade + "位数，您的输入已满。\n请点击确定验证答案或删除重新输入。", Toast.LENGTH_SHORT).show()
            return@OnClickListener
        }

        val str = count_simkeyboard_tv!!.text.toString()

        btnText = str + btnText
        count_simkeyboard_tv!!.text = btnText
        isNotFull = btnText.length <= grade - 1
    }

    private fun nextTest() {
        FileUtils.countDataList.add(countData)
        val intent = Intent(this@CountSimKeyboardActivity, CountGameActivity::class.java)
        intent.putExtra("grade", grade)
        startActivity(intent)
        finish()
    }

    private fun saveAndContinue() {
        var content = randomStr
        if (isRight) {
            content += ";1"
        } else {
            content += ";0"
        }
        if (version == "sound") {
            content += ";1"
        } else {
            content += ";0"
        }
        for (x in result!!) {
            content += ";" + x
        }

        grade++
        if (grade <= maxTestNum) {
            nextTest()
        } else {
            FileUtils.countDataList.add(countData)
            /*val intent = Intent(this@CountSimKeyboardActivity, FeedBackActivity_v2::class.java)
            intent.putExtra("modelName", ModuleHelper.MODULE_COUNT)
            startActivity(intent)
            finish()*/
            // 存储数据，数字模块不需要打分
            val suffix = FileUtils.filePath.substring(FileUtils.filePath.lastIndexOf("."), FileUtils.filePath.length)
            val fileName = "Parkins/" + MD5.md5("${Dua.getInstance().currentDuaUid}${System.currentTimeMillis()}") + suffix

            val data = "{\"score\":\"${0}\"," +
                    "\"doctor\":\"${Dua.getInstance().duaUser.name}\"," +
                    "\"patient\":\"${FileUtils.PATIENT_NAME}\"," +
                    "\"patient_age\":\"${FileUtils.PATIENT_AGE}\"," +
                    "\"patient_sex\":\"${FileUtils.PATIENT_SEX}\"," +
                    "\"patient_med\":\"${FileUtils.PATIENT_MEDICINE}\"," +
                    "\"onoff\":\"${FileUtils.SWITCHING_PERIOD}\"," +
                    "\"filever\":\"${1}\"," +
                    "\"file\":\"${fileName}\"}"
            insertDB(data)
            var filePath = FileUtils.filePath
            // 单项测试
            if (GlobleData.menu_type == ExamPageFragment.MENU_TYPE_SINGLE) {
                startActivity(Intent(this@CountSimKeyboardActivity, UploadActivity::class.java))
                writeData(filePath)
                finish()
            } else {
                AlertDialog.Builder(this@CountSimKeyboardActivity).setTitle("提示").setMessage("即将进入震颤测试").setPositiveButton("确定", DialogInterface.OnClickListener { dialog, which ->
                    // 进入下一项测试，保存数据
                    startActivity(Intent(this, ModelGuideActivity2::class.java))
                    writeData(filePath)
                    finish()
                }).setCancelable(false).show()
            }
        }
    }

    /**
     * 把数据写入文件
     */
    private fun writeData(filePath: String) {
        doAsync {
            var jo = JSONObject()
            jo.put("type", "count")
            jo.put("data", JSON.toJSONString(FileUtils.countDataList))
            val toString = jo.toString()
            println("countData:" + toString)
            FileUtils.writeToFile(toString, filePath)

           // GzipUtil.compressForZip(filePath, filePath + ".gz")
        }
    }

    /**
     * 把数据文件路径插入到数据库
     */
    private fun insertDB(mark: String) {
        /*val historyProvider = HistoryProvider(DataBaseHelper.getInstance(this))
        val history = History(Dua.getInstance().currentDuaId, ModuleHelper.MODULE_COUNT, FileUtils.filePath + ".gz", mark)
        historyProvider.InsertHistory(history)*/
        database.use {
            // 历史数据
            val values = ContentValues()
            values.put(cn.ac.ict.cana.newversion.db.bean.History.BATCH, FileUtils.batch)
            values.put(cn.ac.ict.cana.newversion.db.bean.History.USERID, Dua.getInstance().currentDuaId)
            values.put(cn.ac.ict.cana.newversion.db.bean.History.TYPE, ModuleHelper.MODULE_COUNT)
            values.put(cn.ac.ict.cana.newversion.db.bean.History.FILEPATH, FileUtils.filePath)
            values.put(cn.ac.ict.cana.newversion.db.bean.History.MARK, mark)
            values.put(cn.ac.ict.cana.newversion.db.bean.History.ISUPLOAD, "0")
            // 插入数据库
            insert(cn.ac.ict.cana.newversion.db.bean.History.TABLE_NAME, null, values)
        }
    }

    fun clearText(v: View) {
        count_simkeyboard_tv!!.text = ""
        isNotFull = true
    }

    fun deleteText(v: View) {
        var str = count_simkeyboard_tv!!.text.toString().trim { it <= ' ' }
        if (str == "") {
            return
        }
        str = str.substring(0, str.length - 1)
        count_simkeyboard_tv!!.text = str
        isNotFull = true
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        when (item.itemId) {
            android.R.id.home -> {
                startActivity(Intent(this@CountSimKeyboardActivity, ModelGuideActivity::class.java))
                finish()
            }
        }
        return super.onOptionsItemSelected(item)
    }

    override fun onDestroy() {
        if (pool != null) {
            pool!!.release()
            pool = null
        }

        super.onDestroy()
    }

    fun next(view: View) {
        startActivity(Intent(this, ModelGuideActivity2::class.java))
        finish()
    }

    override fun onBackPressed() {
        startActivity(Intent(this, ModelGuideActivity::class.java))
        finish()
    }
}
